docworks-dts/lib/multiple-files/index.js

```diff
@@ -0,0 +1,47 @@
+const template_ = require('lodash/template')
+const repoCreator = require('./repoCreator')
+const { dtsTripleSlashReference } = require('./generators')
+const {
+	set: setDocumentationGenerator
+} = require('./providers/documentationGenerator')
+
+const {
+	init: initModulesDependencies
+} = require('./providers/modulesDependencies')
+
+const getMainContent = modulesNames =>
+	modulesNames.reduce(
+		(content, name) => content + dtsTripleSlashReference(name) + '\n',
+		''
+	)
+
+const main = (
+	services,
+	{
+		run$wFixer = false,
+		summaryTemplate,
+		ignoredModules = [],
+		mainFileName = 'index.d.ts'
+	} = {}
+) => {
+	const documentationGenerator = item => {
+		if (summaryTemplate) {
+			return template_(summaryTemplate)({ model: item })
+		}
+		return item.summary
+	}
```

Comment:
I think this can be abstracted away from here into the `documentationGenerator` and then you'd have this function be clear:
1. call repoCreator()
2. filter out ignoredModules
3. generateContent

Reply:
agree